### ISSUE TITLE
host header cannot be overriden

### ISSUE THREAD

**Issue by Tuizi** [mcollina/autocannon#162]
I am trying to set the 'host' header, but autocannon always seems to override it.
This request: `autocannon http://localhost:3000/settings -H host=bing -H hostname=google`

```
{ 
  host: 'localhost:3000',
  connection: 'keep-alive',
  hostname: 'google' 
}
```

As you can see the `host` header is set to `localhost:3000` instead of `bing`. As a comparaison it's possible to override the `host` header with Postman.

**Comment by mcollina** [comment]
Would you like to send a PR to add this feature?

**Comment by Tuizi** [comment]
Currently doing the PR with the fix but the tests are failing. Even when I checkout your repo and run tests on the master branch

```
> standard && tap --timeout 45 test/*.test.js

test/argumentParsing.test.js .......................... 8/8
test/cli-ipc.test.js ................................ 13/13
test/cli.test.js .................................... 26/26
test/envPort.test.js ................................ 28/28
test/forever.test.js .................................. 5/5 3s
test/format.test.js ................................... 4/4
test/httpClient.test.js ............................. 65/65
test/httpRequestBuilder.test.js ..................... 17/17
test/onPort.test.js ................................... 0/2 45s
  --on-port flag
  not ok timeout!
    signal: SIGTERM
    handles:
      - type: ChildProcess
        events: []
      - type: Socket
        events:
          - end
          - finish
          - _socketEnd
          - close
      - type: Socket
        events:
          - end
          - finish
          - _socketEnd
          - close
          - data
    expired: TAP
    stack: |
      emit (node_modules/signal-exit/index.js:77:11)
      process.listener (node_modules/signal-exit/index.js:91:7)
      process.emit (node_modules/source-map-support/source-map-support.js:461:21)
      processEmit (node_modules/signal-exit/index.js:155:32)
    test: '--on-port flag'

  --on-port flag
  not ok test count !== plan
    +++ found
    --- wanted
    -26
    +1
    results:
      ok: false
      count: 1
      pass: 0
      fail: 1
      bailout: false
      todo: 0
      skip: 0
      plan:
        start: 1
        end: 26
        skipAll: false
        skipReason: ''
        comment: ''
      failures:
        - ok: false
          id: 1
          name: timeout!
          diag:
            signal: SIGTERM
            handles:
              - type: ChildProcess
                events: []
              - type: Socket
                events:
                  - end
                  - finish
                  - _socketEnd
                  - close
              - type: Socket
                events:
                  - end
                  - finish
                  - _socketEnd
                  - close
                  - data
            expired: TAP
            stack: >
              emit (node_modules/signal-exit/index.js:77:11)

              process.listener (node_modules/signal-exit/index.js:91:7)

              process.emit
              (node_modules/source-map-support/source-map-support.js:461:21)

              processEmit (node_modules/signal-exit/index.js:155:32)
            test: '--on-port flag'
          parent: '--on-port flag'

test/requestIterator.test.js ........................ 22/22
test/run.test.js .................................. 268/268
test/runAmount.test.js .............................. 18/18 9s
test/runRate.test.js .................................. 6/6 4s
total ............................................. 480/482


  480 passing (2m)
  2 failing
```

Any idea?

**Comment by mcollina** [comment]
Are you on Windows perhaps? Tests are passing on Travis (Linux) and on Mac.

Il giorno dom 30 set 2018 alle 02:58 Fabien Rogeret <
notifications@github.com> ha scritto:

**Comment by Tuizi** [comment]
I am on Mac Hight Sierra. Which version of NodeJS and NPM are you using?

**Comment by Tuizi** [comment]
@mcollina https://github.com/mcollina/autocannon/pull/164

**Comment by thekemkid** [comment]
Fixed by #164
